### PR TITLE
add snapshot grouping capability to MCP's snapshot package

### PR DESCRIPTION
**Pull request by ayj**
It's useful to group MCP clients into groups for the purpose of
downstream configuration management. For example, multiple Pilot pods
may share the same desired set of networking API resources. The
snapshot mechanism currently is harded to use the client identifer as
the grouping key. This prevents the use of more advanced grouping
schemes.

This PR introduces GroupIndex which rationalizes the concept of
organizing downstream snapshots into groups.

### CODE REVIEW

File: pkg/mcp/snapshot/snapshot.go [hunk 1]
```diff
@@ -32,20 +32,35 @@ type Snapshot interface {
...
 	mu         sync.RWMutex
 	snapshots  map[string]Snapshot
 	status     map[string]*StatusInfo
 	watchCount int64
+
+	groupIndex GroupIndex
+}
+
+// GroupIndex returns a stable group index for the given MCP client.
+type GroupIndex func(client *mcp.Client) string
```
Comment (ozevren) [inline pkg/mcp/snapshot/snapshot.go]: GroupIndexFn?

File: galley/pkg/runtime/processor.go [hunk 2]
```diff
@@ -180,5 +180,5 @@ func (p *Processor) publish() {
 	sn := p.state.buildSnapshot()
 
 	// TODO: Set the appropriate name for publishing
```
Comment (ozevren) [inline galley/pkg/runtime/processor.go]: Can you remove the stale comment?

File: pkg/mcp/snapshot/snapshot.go [hunk 3]
```diff
@@ -174,37 +188,37 @@ func (c *Cache) SetSnapshot(node string, snapshot Snapshot) {
...
 }
 
-// Status returns informational status for a client.
-func (c *Cache) Status(node string) *StatusInfo {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-	if info, ok := c.status[node]; ok {
+// Status returns informational status for a group.
+func (c *Cache) Status(group string) *StatusInfo {
+	c.mu.Lock()
```
Comment (ozevren) [inline pkg/mcp/snapshot/snapshot.go]: Accidental change to RW lock mode?

Comment (ayj) [reply pkg/mcp/snapshot/snapshot.go]: Looks like a bad merge. I'll fix.

**Comment by ayj** [comment]
cc @kyessenov for approving [one-line mixer change](https://github.com/istio/istio/pull/9041/files#diff-6855a9bea0623064fd9c7a3391e1a952R47)

**Pull request merged**